ci/cleanup-pr-previews

```diff
@@ -0,0 +1,247 @@
+
+This script fetches all pr-XXXXX folders from docs/pr-preview/ in the gh-pages branch,
+checks PR status via GitHub API, and removes folders for closed/merged/deleted PRs.
+
+USAGE: $0 [OPTIONS]
+
+OPTIONS:
+    -n, --dry-run    Preview what would be deleted without actually deleting
+    --push           Commit and push changes to gh-pages (default: false, requires manual push)
+    -h, --help       Show this help message
+
+EXAMPLES:
+    $0 -n                     # Preview what would be cleaned up (RECOMMENDED FIRST)
+    $0                        # Clean up folders locally (no push)
+    $0 --push                 # Clean up folders and push to gh-pages branch
+    $0 --dry-run --push       # Invalid combination (dry-run takes precedence)
+
+REQUIREMENTS:
+    - GH_TOKEN environment variable must be set with appropriate permissions
+    - 'gh' (GitHub CLI) must be installed and authenticated
+    - 'jq' must be installed for JSON parsing
+
+SAFETY:
+Always run with --dry-run first to verify expected behavior before actual cleanup.
+The script will show a summary of what would be removed. Use --push to automatically
+commit and push changes, otherwise manual git operations are required.
+
+This script is specifically designed for the NVIDIA/cuda-python repository structure.
+EOF
+    exit 1
+}
+
+# Configuration - hardcoded for this specific repository
+REPOSITORY="NVIDIA/cuda-python"
+DRY_RUN="false"
+PUSH_CHANGES="false"
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        -n|--dry-run)
+            DRY_RUN="true"
+            shift
+            ;;
+        --push)
+            PUSH_CHANGES="true"
+            shift
+            ;;
+        -h|--help)
+            usage
+            ;;
+        *)
+            echo -e "${RED}[ERROR]${NC} Unknown option: $1" >&2
+            echo "Use --help for usage information" >&2
+            exit 1
+            ;;
+    esac
+done
+
+# Validate required tools and environment
+echo -e "${YELLOW}[INFO]${NC} Checking prerequisites..."
+
+if [[ -z "${GH_TOKEN:-}" ]]; then
+    echo -e "${RED}[ERROR]${NC} GH_TOKEN environment variable is required" >&2
+    exit 1
+fi
+
+if ! command -v jq >/dev/null 2>&1; then
+    echo -e "${RED}[ERROR]${NC} jq is required but not installed" >&2
+    exit 1
+fi
+
+if ! command -v gh >/dev/null 2>&1; then
+    echo -e "${RED}[ERROR]${NC} GitHub CLI (gh) is required but not installed" >&2
+    exit 1
+fi
+
+echo -e "${GREEN}[INFO]${NC} All prerequisites satisfied"
+
+# Fetch PR preview folders from gh-pages branch
+echo -e "${YELLOW}[INFO]${NC} Fetching PR preview folders from gh-pages branch..."
+
+# Get the list of pr-XXXXX folders from gh-pages branch
+PR_FOLDERS=$(gh api repos/"${REPOSITORY}"/contents/docs/pr-preview \
+    --header "Accept: application/vnd.github+json" \
+    --jq '.[] | select(.type == "dir" and (.name | test("^pr-[0-9]+$"))) | .name' \
+    --field ref=gh-pages 2>/dev/null || true)
```

Comment:
This does not work. I see locally that `PR_FOLDERS` is populated with the following content:
```
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest",
  "status": "404"
}
```
Please fix.